### ISSUE TITLE
Support for Two Pass FFMPeg Options

### ISSUE THREAD

**Issue by ZachMyers3** [HaveAGitGat/Tdarr_Plugins#146]
Hello,

For best quality transcoding VP9 suggests [a two pass approach](https://trac.ffmpeg.org/wiki/Encode/VP9). With the current plugin configuration I do not see a way to do this.

It would be nice to be able to do something like this in the future, let me know if you need any more information.

Thanks

**Comment by ZachMyers3** [comment]
Notably, I'd like to pass my file as input and ```-f null /dev/null``` as output, but both input and output are tied to the ```file.file``` properties and I don't see a way around this.

**Comment by ZachMyers3** [comment]
I'm back after working on this for the past couple weeks and I believe I have a better understanding of what needs to be done to make this possible.

Within `ffmpeg` I'm able to pass a logfile with `-passlogfile`. This is great! The existence of this log file can also determine if I'm in my first pass or second pass when encoding so I can dynamically assign `-pass 1` or `-pass 2`. The issue I'm now running into however is that I am unable to reference the original source file on the second pass.

For example, I'm encoding `myfile.mp4` into `myfile.webm` in VP9 format. Tdarr begins transcoding `myfile.mp4` on the first pass of this application. `file._id` `file.file` and `file.meta.SourceFile` all reference `/path/to/videodir/myfile.mp4`. When this transcode is finished `file._id` `file.file` and `file.meta.SourceFile` now all reference `/path/to/cache/myfile-hashvalue.mp4` and there is no reference to `myfile.mp4`. Because I can no longer reference the original file, I cannot transcode properly as the file create on the first run is corrupt as the primary goal for the first pass is to generate the log file.

To sum it up, if there was some attribute within the `file` object that _always_ pointed to the original file, I believe it would be possible to do two-pass encoding within Tdarr.

Thanks for the help, I can try to provide any more input if necessary.

**Comment by HaveAGitGat** [comment]
Hi, you can access the original file using the `otherArguments.originalLibraryFile` object, example here:

https://github.com/HaveAGitGat/Tdarr_Plugins/blob/a3ce1cf21b198a8a07c375449bc34499507bb58c/Community/Tdarr_Plugin_rr01_drpeppershaker_extract_subs_to_SRT.js#L78

2-encoding is bit a tricky.  I think some users have it running using various workarounds including encoding metadata into the file so it's possible to know which pass the file is on.

**Comment by schklom** [comment]
@ZachMyers3 Can you share more details on how you accomplished this?
I'm discovering Tdarr, and I have no idea how to do what you did.

**Comment by ZachMyers3** [comment]
@schklom Sorry to bring bad news, but I away from two pass and just applied VP9 via one pass. I've actually pushed this into the community [here](https://github.com/HaveAGitGat/Tdarr_Plugins/blob/master/Community/Tdarr_Plugin_VP92_VP9_Match_Bitrate_One_Pass.js).

Sorry I couldn't help you more on it.
